File: examples/perf_test.py
# Sample Python Polars code to compare with the `perf_test.rs` example. The
# following is based on code from Alexander Beedie in the following GitHub
# thread: https://github.com/pola-rs/polars/issues/5568#issuecomment-1526316286

from codetiming import Timer
from datetime import date
import polars as pl

# Quickly spin-up a 1,000,000 element DataFrame.
df = pl.DataFrame(
    {
        "Int": range(250_000),
        "Float": 123.456789,
        "Date": date.today(),
        "String": "Test"
    }
)

# Export to Excel from polars.
with Timer():
    df.write_excel("dataframe_pl.xlsx")

# Export to Excel from pandas.
pf = df.to_pandas()
with Timer():
    pf.to_excel("dataframe_pd.xlsx", index=False)

File: examples/perf_test.rs
// SPDX-License-Identifier: MIT OR Apache-2.0
//
// Copyright 2023-2024, John McNamara, jmcnamara@cpan.org

//! Simple performance test to compare with the Python Polars example in
//! `perf_test.py`.

use chrono::prelude::*;
use polars::prelude::*;
use polars_excel_writer::PolarsXlsxWriter;
use std::time::Instant;

const DATA_SIZE: usize = 250_000;

fn main() {
    // Create a sample dataframe for testing.
    let df: DataFrame = df!(
        "Int" => &[1; DATA_SIZE],
        "Float" => &[123.456789; DATA_SIZE],
        "Date" => &[Utc::now().date_naive(); DATA_SIZE],
        "String" => &["Test"; DATA_SIZE],
    )
    .unwrap();

    let timer = Instant::now();
    example(&df).unwrap();
    println!("Elapsed time: {:.2?}", timer.elapsed());
}

fn example(df: &DataFrame) -> PolarsResult<()> {
    let mut xlsx_writer = PolarsXlsxWriter::new();
    xlsx_writer.write_dataframe(df)?;
    xlsx_writer.save("dataframe_rs.xlsx")?;

    Ok(())
}
